incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -986,18 +1624,21 @@ private void doEnd(
         long traceId,
         long authorization)
     {
-        final EndFW end = endRW.wrap(writeBuffer, 0, writeBuffer.capacity())
-            .originId(originId)
-            .routedId(routedId)
-            .streamId(streamId)
-            .sequence(sequence)
-            .acknowledge(acknowledge)
-            .maximum(maximum)
-            .traceId(traceId)
-            .authorization(authorization)
-            .build();
+        if (receiver != null)
```

Comment:
This defensive coding seems like it is hiding a different bug.
Why would `receiver` be `null` here?

Reply:
TBH, I'm not sure why it was needed. I just deleted it, and everything still seems to work fine